test/integration/integration_test.cc

```diff
@@ -477,6 +478,7 @@ name: matcher
 
     ASSERT_TRUE(response->waitForEndStream());
     EXPECT_THAT(response->headers(), HttpStatusIs("200"));
+    codec_client_->close();
```

Comment:
nit: the `codec_client_->close()` on line 493 is moot.

Question: What is the purpose of the second codec, after the `codec_client_` is already closed? will the rest of this test just be the same as using `codec_client_` instead of `second_codec`?